### PR TITLE
faFormat MultiFaNoGapBed

### PR DESCRIPTION
**Pull request by rimangan**
# Description
faFormat already has a 'noGap' option, which allows the user to make a bed of all ungapped regions in the sequence. This PR extends this functionality to multiFa format alignments. In this use case, we can make a 'noGap' bed for any of the aligned query sequences in a multiFa. To accomplish this, the user must specify two other options: chromName (the name of the chromosome associated with the multiFa file), and querySeqName (the name of the fasta record for which you want to generate a noGap bed file). 
The algorithm to calculate this in bed/info.go is nearly the same as for the original noGap option, with some added logic to handle refPos/alnPos conversions and bookkeeping. 

## Relevant Links

- [Display Text](https://www.vertgenlab.org/)

### Checklist before requesting a review

- [ ] I performed a self-review of my code
- [ ] If it this a core feature, I have added thorough tests
- [ ] The command `go fmt` or `make clean` was used on all files included

### Testing

None

### Screenshots & Media

None

### Edge cases / Breaking Changes / Known Issues

None

### CODE REVIEW

**Review by edotau — DISMISSED** [review]

File: bed/info.go [hunk 1]
```diff
@@ -2,11 +2,53 @@ package bed
...
+		log.Fatalf("Error: seqName: %s, not found in records.\n", seqName)
+	}
+
+	for currAlnPos = 0; currAlnPos < len(records[0].Seq); currAlnPos++ {
+		if dna.DefineBase(records[seqNameIndex].Seq[currAlnPos]) && !inRegion {
+			inRegion = true
+			startRefPos = fasta.AlnPosToRefPosCounter(records[0], currAlnPos, lastRefPos, lastAlnPos)
+			lastRefPos, lastAlnPos = startRefPos, currAlnPos
+		} else if !(dna.DefineBase(records[seqNameIndex].Seq[currAlnPos])) && inRegion {
+			endRefPos = fasta.AlnPosToRefPosCounter(records[0], currAlnPos, lastRefPos, lastAlnPos)
```
[the remark this answers is not in the document]
Comment (RavenYL) [reply bed/info.go:36]: I think it shouldn't because in the multifa each sequence's index is by AlnPos not RefPos. When it's a gap in the reference sequence, sequence[alnPos] stores the value "dna.Gap"

[the remark this answers is not in the document]
Comment (RavenYL) [reply bed/info.go:36]: I think when alnPos is a gap in reference sequence, the ref position to the right (the first position to the first that is not a gap) is reported
```
func AlnPosToRefPosCounterSeq(record []dna.Base, AlnPos int, refStart int, alnStart int) int {
	if alnStart > AlnPos {
		refStart, alnStart = 0, 0 //in case the alnStart was improperly set (greater than the desired position, we reset the counters to 0.
	}
	for t := alnStart; t < AlnPos; t++ {
		if t == len(record) {
			log.Fatalf("Ran out of chromosome.")
		} else if record[t] != dna.Gap {
			refStart++
		}
	}
	return refStart
}
```

File: bed/info.go [hunk 2]
```diff
@@ -2,11 +2,53 @@ package bed
...
+	var inRegion bool = false
+	var startRefPos, endRefPos int = 0, 0
+	var lastRefPos, lastAlnPos = 0, 0
+	var currAlnPos int = 0
+
+	//first we find the index of seqName
+	for i := 0; i < len(records); i++ {
+		if records[i].Name == seqName {
+			seqNameIndex = i
+			break
```
[the remark this answers is not in the document]
Comment (rimangan) [reply bed/info.go:22]: nice catch! Will fix in next commit

**Review by craiglowe — APPROVED** [review]
Great!

**Review by RavenYL — APPROVED** [review]
